### PR TITLE
Outstanding tree: keyboard navigation for autocomplete results

### PR DESCRIPTION
**Pull request by modernserf**
## Links
* Remix link https://outstanding-tree.glitch.me/
* Manuscript link https://glitch.manuscript.com/f/cases/3328306/

## GIF/Screenshots:
autocomplete nav 2019-04-15 18_33_23

## Changes:
* Move autocomplete logic to search form
* track keydown events on search input
* refactor search state to minimize wasted renders

## How To Test:
* visit https://outstanding-tree.glitch.me/secret and enable algolia search
* use the search bar, and navigate with up/down arrow keys

## Feedback I'm looking for:
* does this work the way you would expect keyboard navigation to work?
* what should the active state for the autocomplete items look like?

### CODE REVIEW

**Comment by keithk** [comment]
Going to run through the code now, but opened this and tabbed through and it worked great, and worked with voiceover open as well 🎉

File: src/components/search/autocomplete.styl [hunk 1]
```diff
@@ -34,6 +26,9 @@
   padding: 6px 12px
   &:hover, &:focus
     background-color: secondary-background-hover
+  &.selected
+    background-color: #339
```
Comment (ttseng) [inline src/components/search/autocomplete.styl]: can use the `general-link` variable in `constants.styl` (#244776)

File: src/components/search/autocomplete.styl [hunk 2]
```diff
@@ -34,6 +26,9 @@
   padding: 6px 12px
   &:hover, &:focus
     background-color: secondary-background-hover
```
Comment (ttseng) [inline src/components/search/autocomplete.styl]: let's make the hover, focus, and selected styling consistent
```
&:hover, &:focus, &.selected
    background-color: general-link
    color: white

**Comment by ttseng** [comment]
The tab navigation is working really well.  
I'm finding with tab navigation, I can select the "See all results" link, but with the up + down arrows, the "See all results" link isn't selectable:
arrow-nav

**Comment by modernserf** [comment]
- normalize styles between focus, hover, arrow selection
- make 'see all results' arrow-selectable

**Comment by ttseng** [comment]
I'm seeing this error when using keyboard navigation + enter on the See All Results
`index.js:3671 TypeError: urlForItem[selectedResult.type] is not a function`

**Comment by modernserf** [comment]
@ttseng that should be fixed now.

**Pull request merged**